src/RepoDriver.sol

```diff
@@ -0,0 +1,508 @@
+    /// @return payload The AnyApi request payload.
+    function _requestPayload(Forge forge, bytes memory name)
+        internal
+        view
+        returns (bytes memory payload)
+    {
+        // slither-disable-next-line uninitialized-local
+        BufferChainlink.buffer memory buffer;
+        buffer = BufferChainlink.init(buffer, 256);
+        buffer.encodeString("get");
+        buffer.encodeString(_requestUrl(forge, name));
+        buffer.encodeString("path");
+        buffer.encodeString(_requestPath());
+        return buffer.buf;
+    }
+
+    /// @notice Builds the URL for fetch the `FUNDING.json` file for the given repository.
+    /// @param forge The forge where the repository is stored.
+    /// @param name The name of the repository.
+    /// @return url The built URL.
+    function _requestUrl(Forge forge, bytes memory name)
+        internal
+        pure
+        returns (string memory url)
+    {
+        if (forge == Forge.GitHub) {
+            return string.concat(
+                "https://raw.githubusercontent.com/", string(name), "/HEAD/FUNDING.json"
+            );
+        } else if (forge == Forge.GitLab) {
+            return string.concat("https://gitlab.com/", string(name), "/-/raw/HEAD/FUNDING.json");
+        } else {
+            revert("Unsupported forge");
+        }
+    }
+
+    /// @notice Builds the JSON path inside `FUNDING.json` where the repository owner is stored.
+    /// @return path The comma-separated JSON path.
+    function _requestPath() internal view returns (string memory path) {
+        // slither-disable-next-line uninitialized-local
+        string memory chainName;
+        if (block.chainid == CHAIN_ID_ETHEREUM) chainName = CHAIN_NAME_ETHEREUM;
+        else if (block.chainid == CHAIN_ID_GOERLI) chainName = CHAIN_NAME_GOERLI;
+        else revert("Unsupported chain");
+        return string.concat("drips,", chainName, ",ownedBy");
+    }
+
+    /// @notice Updates the repository owner. Callable only by the AnyApi operator.
+    /// @param requestId The ID of the AnyApi request.
+    /// Must be the same as the request ID generated when requesting an owner update,
+    /// this function will update the repository that was requested back then.
+    /// @param ownerRaw The new owner of the repository. Must be a 20 bytes long address.
+    function updateRepoOwnerByAnyApi(bytes32 requestId, bytes calldata ownerRaw) public {
+        RepoDriverAnyApiStorage storage storageRef = _repoDriverAnyApiStorage();
+        require(msg.sender == address(storageRef.operator), "Callable only by the operator");
+        uint224 repoId = storageRef.requestedUpdates[requestId];
+        require(repoId != 0, "Unknown request ID");
+        delete storageRef.requestedUpdates[requestId];
+        require(ownerRaw.length == 20, "Invalid owner length");
+        address owner = address(bytes20(ownerRaw));
+        _repoDriverStorage().repoOwners[repoId] = owner;
```

Comment:
One idea. Together, with the `owner` we also store a `lastUpdated` and add some delay period. 

An update is only possible once every 30 minutes or something.

Reply:
What would it protect from? It certainly would be annoying, people will be making mistakes and many of them will be requesting multiple updates in the span of minutes when they're doing the initial configuration.

Reply:
Yeah that's a good point.